blobfuse/blobfuse.cpp

```diff
@@ -535,12 +546,60 @@ int set_log_mask(const char * min_log_level_char)
         return 0;
     }
 
-    syslog(LOG_CRIT, "Unable to start blobfuse. Error: Invalid log level \"%s\"", min_log_level.c_str());
-    fprintf(stderr, "Error: Invalid log level \"%s\".  Permitted values are LOG_OFF, LOG_CRIT, LOG_ERR, LOG_WARNING, LOG_INFO, LOG_DEBUG.\n", min_log_level.c_str());
-    fprintf(stdout, "If not specified, logging will default to LOG_WARNING.\n\n");
+    if (blobfuseInit) {
+        syslog(LOG_CRIT, "Unable to start blobfuse. Error: Invalid log level \"%s\"", min_log_level.c_str());
+        fprintf(stderr, "Error: Invalid log level \"%s\".  Permitted values are LOG_OFF, LOG_CRIT, LOG_ERR, LOG_WARNING, LOG_INFO, LOG_DEBUG.\n", min_log_level.c_str());
+        fprintf(stdout, "If not specified, logging will default to LOG_WARNING.\n\n");
+    } else {
+        set_log_mask(options.log_level, false);
+    }
     return 1;
 }
 
+int refresh_from_config_file(const std::string configFile)
+{
+    std::ifstream file(configFile);
```

Comment:
If you meant call the original read file then I avoided that intentionally. As the original read file reads all the config parameter from file and updates the global variables, if user by mistake did any other change then it may bring down the running instance. So just reading logging level when the signal arrives.

Reply:
// add comments to the method that it will only be called on refresh and it will only refresh log_level.  Change the